packages/shared/lib/core/utils/object.ts

```diff
@@ -10,30 +10,33 @@
 export function migrateObjects<T>(oldObj: T, newObj: T): T {
     /* eslint-disable @typescript-eslint/no-explicit-any */
     const _helper = (curObj: any, oldObj: any, newObj: any): any => {
-        // Iterate through each key of a new object...
-        for (const k in newObj) {
+        // Iterate through each key of a new AND old objects...
+        for (const key in { ...oldObj, ...newObj }) {
```

Comment:
with this change, properties which are in `oldObj` but not in `newObj` are not deleted